### PR TITLE
feat: rank hparams with hp importance [DET-5105]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Opportunistically rank the available hparams with hp importance if available.

HP picker without hp importance:
Screen Shot 2021-03-25 at 11 58 22 AM

HP picker with hp importance:
Screen Shot 2021-03-25 at 12 28 40 PM

## Test Plan

run the following experiment:
```
det e create ./e2e_tests/tests/fixtures/mnist_pytorch/random.yaml ./examples/tutorials/mnist_pytorch
```

When the hp importance is available the HP picker shows the hp importance value next to each hparam and orders the hp by the importance value. Otherwise it doesn't show the value and shows the hps alphabetically.

the response for hp importance:
```
{
  "result": {
    "trainingMetrics": {
      "loss": {
        "hpImportance": {
          "dropout1": 1.1933941110928465e-18,
          "dropout2": 2.0018029756745813e-26,
          "learning_rate": 7.802653352931384e-30,
          "n_filters1": 0.4068482249453416,
          "n_filters2": 0.4735429929594936
        },
        "experimentProgress": 1,
        "error": "not enough trials for HP importance: 33",
        "pending": false,
        "inProgress": false
      }
    },
    "validationMetrics": {
      "validation_loss": {
        "hpImportance": {
          "dropout1": 0.0000694575149247851,
          "dropout2": 2.1103457611012306e-22,
          "learning_rate": 8.019502373685075e-32,
          "n_filters1": 0.9966329322317776,
          "n_filters2": 0.4015763949920965
        },
        "experimentProgress": 1,
        "error": "not enough trials for HP importance: 32",
        "pending": false,
        "inProgress": false
      }
    }
  }
}
```

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by hkang1** [comment]
@mackrorysd let me know if the proto change doesn't work for you.

File: proto/src/determined/api/v1/experiment.proto [hunk 1]
```diff
@@ -489,6 +489,9 @@ message GetHPImportanceRequest {
 
 // Response to GetHPImportanceRequest
 message GetHPImportanceResponse {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "training_metrics", "validation_metrics" ] }
```
Comment (mackrorysd) [inline proto/src/determined/api/v1/experiment.proto:493]: Can't comment well on the rest of the PR, but this part looks good to me.

File: webui/react/src/components/Section.tsx [hunk 2]
```diff
@@ -40,12 +42,15 @@ const Section: React.FC<Props> = (props: PropsWithChildren<Props>) => {
...
-      {props.filters && (
+      {!props.loading && props.filters && (
         <div className={css.filterBar}>
           {props.filters}
         </div>
       )}
-      <div className={css.body}>{props.children}</div>
+      <div className={css.body}>
+        {!props.loading && props.children}
+        {props.loading && <Spinner />}
```
Comment (apizzini) [inline webui/react/src/components/Section.tsx:52]: love that, super-minor, but looks to be not centered?

Schermata 2021-03-26 alle 09 39 27

**Review by apizzini — APPROVED** [review]
this looks very good :)

File: webui/react/src/components/Section.tsx (hunk 2 above)
Comment (hkang1) [reply webui/react/src/components/Section.tsx]: good eye! Should be good to go now.

**Comment by hkang1** [comment]
> this looks very good :)

Thanks for the speedy review!
